--- lib/bytebuffer.rb.orig	2024-07-20 14:25:37
+++ lib/bytebuffer.rb	2024-07-20 14:26:17
@@ -178,11 +178,14 @@
 
   # Read a single 8-bit unsigned integer from the {ByteBuffer}.
   # @return [Integer] the read value.
-  def read_u8; ByteBufferExtension.read_u8(@ptr); end
+  def read_u8; ByteBufferExtension.read_u8(@ptr) & 0xff; end
 
   # Read a single 8-bit signed integer from the {ByteBuffer}.
   # @return [Integer] the read value.
-  def read_i8; ByteBufferExtension.read_i8(@ptr); end
+  def read_i8
+    v=ByteBufferExtension.read_i8(@ptr)
+    (v&(0xff/2))-(v&(0xff/2+1))
+  end
 
   # Read a single 16-bit unsigned integer from the {ByteBuffer}.
   # @return [Integer] the read value.
@@ -190,7 +193,10 @@
 
   # Read a single 16-bit signed integer from the {ByteBuffer}.
   # @return [Integer] the read value.
-  def read_i16; ByteBufferExtension.read_i16(@ptr); end
+  def read_i16
+    v=ByteBufferExtension.read_i16(@ptr)
+    (v&(0xffff/2))-(v&(0xffff/2+1))
+  end
 
   # Read a single 32-bit unsigned integer from the {ByteBuffer}.
   # @return [Integer] the read value.
